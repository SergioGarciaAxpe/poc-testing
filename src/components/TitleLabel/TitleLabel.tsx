
type TitleLabelProps = {
    title: string;
    subtitle: string;
}

export const TitleLabel = ({title, subtitle}: TitleLabelProps) => {
  return (
    <div>
        <h1>{title}</h1>
        <h3>{subtitle}</h3>
    </div>
  )
}
