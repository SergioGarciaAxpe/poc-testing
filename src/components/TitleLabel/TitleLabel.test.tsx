import { render } from "@testing-library/react"
import '@testing-library/jest-dom';
import { TitleLabel } from "./TitleLabel"

describe('Testing <TitleLabel />', () => {
    test('Render <TitleLabel />', () => {
        const title = 'Hola Mundo';
        const subtitle = 'Mi nombre es Sergio';
        const { asFragment, getByText } = render(<TitleLabel title={title} subtitle={subtitle} />);
        expect(getByText('Hola Mundo')).toBeInTheDocument()
        expect(getByText('Mi nombre es Sergio')).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot(`
            <div>
                <h1>Hola Mundo</h1>
                <h3>Mi nombre es Sergio</h3>
            </div>
        `)
    })
})