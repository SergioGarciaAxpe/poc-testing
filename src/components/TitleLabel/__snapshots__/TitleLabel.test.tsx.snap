// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TitleLabel /> Render <TitleLabel />: 
            <div>
                <h1>Hola Mundo</h1>
                <h3>Mi nombre es Sergio</h3>
            </div>
         1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hola Mundo
    </h1>
    <h3>
      Mi nombre es Sergio
    </h3>
  </div>
</DocumentFragment>
`;
